sycl/doc/extensions/SPIRV/SPV_INTEL_debug_module.asciidoc

```diff
@@ -0,0 +1,206 @@
+
+Status
+------
+
+First draft
+
+Version
+-------
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | 2021-06-22
+| Revision           | 3
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.5 Revision 5, Unified
+
+This extension requires SPIR-V 1.0. and OpenCL.DebugInfo.100 Extended
+Instruction Set
+
+Overview
+--------
+
+This extension adds new instruction `DebugModuleINTEL` to OpenCL.DebugInfo.100
+Extended Instruction Set. This instruction allows to represent concept of a
+"module" used in some programming languages.
+Example - Fortran module or Clang module. The proposed instruction is defined in
+accordance with LLVM `!DIModule` metadata for easier translation between
+SPIR-V/LLVM.
+
+Extension Name
+--------------
+
+To use this extension within a SPIR-V module, the following
+*OpExtension* must be present in the module:
+
+----
+OpExtension "SPV_INTEL_debug_module"
+----
+
+New Capabilities
+----------------
+This extension introduces a new capability:
+
+----
+DebugInfoModuleINTEL
+----
+
+New Instructions
+----------------
+This extension adds the following extended instruction:
+
+----
+DebugModuleINTEL
+----
+
+Token Number Assignments
+------------------------
+[width="45%",cols="30,15"]
+|===============================
+| DebugInfoModuleINTEL | 6114
+| DebugModuleINTEL | 36
+|===============================
+
+Modifications to the SPIR-V Specification, Version 1.5, Revision, Unified
+--------------------------------------------------------------------------
+
+Capabilities
+~~~~~~~~~~~~
+
+Modify Section 3.31, "Capability", adding these rows to the Capability table:
+
+--
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 6114 | *DebugInfoModuleINTEL* +
+Allow to use `DebugModuleINTEL` extended instruction if OpenCL.DebugInfo.100 Extended Instruction Set is imported |
+|====
+--
+
+Modifications to the OpenCL.DebugInfo.100 Extended Instruction Set
+------------------------------------------------------------------
+
+Instructions
+~~~~~~~~~~~
+
+Modify section 4.10 "Imported Entities", adding the following instruction:
+
+[cols="14", width="100%"]
+|=====
+13+^|*DebugModuleINTEL* +
+
+Represents a module in the programming language, for example a Fortran module. +
+
+_Result Type_ must be *OpTypeVoid* +
+
+_Set_ must be result of an OpExtInstImport instruction. +
+
+_Name_ is an *OpString*, holding the name of the imported module. +
+
+_Source_ is a *DebugSource* instruction representing text of the source program +
+of the module. +
+
+_Line_ is a single word literal denoting the source line number at which the +
+declaration or use of a module appears in the _Source_. +
+
+_Parent_ is id of a debug instruction which represents the parent lexical scope. +
+
+_ConfigurationMacros_ is an *OpString*, holding space-separated shell-quoted +
```

Comment:
I'm not sure we need ` +` at the end of every line. I think ` +` at the end of every section/paragraph should be enough.

Reply:
Actually I'm sure we don't need that many ` +`. The rendered spec looks inconsistent: https://github.com/intel/llvm/blob/d901a275261d7b491a369e0387c9e9e93a47b3fc/sycl/doc/extensions/SPIRV/SPV_INTEL_debug_module.asciidoc#instructions 

Operands are kind of aligned to the left size, while the last paragraphs are distributed over the width of the table.